ext/js/language/translator.js

```diff
@@ -236,49 +248,153 @@ export class Translator {
         return {dictionaryEntries, originalTextLength};
     }
 
+    /**
+     *
+     * @param {import('dictionary').TermDictionaryEntry} existingEntry
+     * @param {import('dictionary').InflectionHypothesis[]} inflectionHypotheses
+     */
+    _mergeInflectionHypotheses(existingEntry, inflectionHypotheses) {
+        const existingHypotheses = existingEntry.inflectionHypotheses;
+
+        inflectionHypotheses.forEach(({source, inflections}) => {
+            const duplicate = existingHypotheses.find((hypothesis) => this._areInflectionHyphothesesEqual(hypothesis.inflections, inflections));
+            if (!duplicate) {
+                existingEntry.inflectionHypotheses.push({source, inflections});
+            } else if (duplicate.source !== source) {
+                duplicate.source = 'both';
+            }
+        });
+    }
+
+    /**
+     * @param {import('dictionary').Inflections} hypothesis1
+     * @param {import('dictionary').Inflections} hypothesis2
+     * @returns {boolean}
+     */
+    _areInflectionHyphothesesEqual(hypothesis1, hypothesis2) {
+        const set1 = new Set(hypothesis1);
+        const set2 = new Set(hypothesis2);
+
+        return set1.size === set2.size && [...set1].every((x) => set2.has(x));
```

Comment:
This is probably more efficient if you only construct 1 set and no temporary array. Also do the size comparison before any of that for an earlier exit.

Reply:
On closer inspection, what I actually need here is something like 
```js
DictionaryDataUtil.areArraysEqual(
    [...hypothesis.inflections].sort(),
    [...inflections].sort()
));
```
is that ok?

Reply:
I'd say that's probably still more inefficient than it needs to be, esp. given that these arrays are probably likely to just be 0-2 long in the normal case. I'd suggest creating some utility function in core or a new core/*.js file that's like `areArraysEqualIgnoreOrder` and just do a 2-deep for loop.

```js
export function areArraysEqualIgnoreOrder(array1, array2) {
    const {length} = array1;
    if (length !== array2.length) { return false; }
    for (let i = 0; i < length; ++i) {
        if (!array2.includes(array1[i])) { return false; }
    }
    return true;
}
```

Rationale: for the expected use case of very small arrays, just doing what an in-place check is likely faster than reconstructing a new object, be it a set or a sorted array. I've always had some emphasis on keeping the translator fairly efficient, which is driving this feedback.

Reply:
This is an edge case, not sure if it might come up for some language, but  `areArraysEqualIgnoreOrder(["a", "b", "b"], ["a", "a", "b"])` is `true`.

Reply:
Actually, this seems to be the most performant implementation:
```js
   function areArraysEqualIgnoreOrder(array1, array2) {
        if (array1.length !== array2.length) {
            return false;
        }
    
        const frequencyCounter = {};
    
        for (const element of array1) {
            frequencyCounter[element] = (frequencyCounter[element] || 0) + 1;
        }
    
        for (const element of array2) {
            if (!frequencyCounter[element]) {
                return false;
            }
            frequencyCounter[element]--;
        }
       
        return true;
    }  
```

Reply:
How does that compare to using a map instead of `{}`?

Reply:
That implementation also only works for strings and has some edge cases related to `Object.prototype`, e.g. an element named `toString` will break it.

Reply:
I had made a basic [benchmark](https://jsbench.me/r8lr29janj/1) and it kept telling me Map is slower, but that might be an artifact of the setup. Map does seem better for this.

Reply:
While I do advocate for performance considerations where appropriate, I wouldn't want to sacrifice good code in favor of it; IMO object-as-a-map is not good modern JS. (I mentioned my thought about that in [point 1 here](https://github.com/themoeway/yomitan/issues/342#issue-2029729570).)

Reply:
Another question that comes to mind: are arrays with repeats valid for your use case? Like should those ever be generated? I know there are several situations where the extension generates an "array", but implicitly each item is unique so it's only an "array" for serialization purposes.

If each element is expected to be unique, which I think was my original assumption, you can probably simplify the implementation (e.g. change Map to Set). If not, I think the length equality check needs removal if you want to consider `["a", "b", "b"] = ["a", "a", "b"]` and `["a", "b", "b"] = ["a", "b"]`.